codegen/protocol-test-codegen-local/Tests/swift-codegen/WaitersTests/ErrorTypeMatcherTests.swift

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import Waiters
+@testable import ClientRuntime
+
+#if swift(>=5.7)
+
+class ErrorTypeMatcherTests: XCTestCase {
+
+    // expected errorType for these tests: "MyError"
+
+    // MARK: - errorType matcher
+
+    func test_errorType_matchesWhenErrorTypeMatchesAndErrorIsAWaiterTypedError() async throws {
+        let error = WaiterTypedErrorImpl()
+        let subject = try WaitersClient.errorTypeMatcherWaiterConfig().acceptors[0]
+        let match = subject.evaluate(input: anInput, result: .failure(error))
+        XCTAssertEqual(match, .success(.failure(error)))
+    }
+
+    func test_errorType_doesNotMatchWhenErrorTypeDoesNotMatchAndErrorIsAWaiterTypedError() async throws {
+        let error = WaiterTypedErrorNoMatchImpl()
+        let subject = try WaitersClient.errorTypeMatcherWaiterConfig().acceptors[0]
+        let match = subject.evaluate(input: anInput, result: .failure(error))
+        XCTAssertNil(match)
+    }
+
+    func test_errorType_doesNotMatchWhenErrorTypeMatchesButErrorIsNotAWaiterTypedError() async throws {
+        let error = WaiterTypedErrorNoMatchImpl()
+        let subject = try WaitersClient.errorTypeMatcherWaiterConfig().acceptors[0]
+        let match = subject.evaluate(input: anInput, result: .failure(error))
+        XCTAssertNil(match)
+    }
+
+    func test_errorType_doesNotMatchWhenResultIsSuccess() async throws {
+        let response = GetWidgetOutputResponse()
+        let subject = try WaitersClient.errorTypeMatcherWaiterConfig().acceptors[0]
+        let match = subject.evaluate(input: anInput, result: .success(response))
+        XCTAssertNil(match)
+    }
+}
+
+// Error types used in tests above
+
+private struct WaiterTypedErrorImpl: WaiterTypedError, Equatable {
```

Comment:
While wordy, its preferable to use `Implementation` here over `Impl` for clarity.
From https://www.swift.org/documentation/api-design-guidelines/#promote-clear-usage
>Avoid abbreviations. Abbreviations, especially non-standard ones, are effectively terms-of-art, because understanding depends on correctly translating them into their non-abbreviated forms.

Reply:
lol time to give up on `Int` and `Bool` then

Reply:
Renamed these errors without abbreviations (and with more descriptive names too)